### PR TITLE
Added Category and endpoints for them

### PR DESCRIPTION
**Pull request by IhorTrokhymchuk**

### CODE REVIEW

File: src/main/java/com/example/store/model/Category.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Data
+@NoArgsConstructor
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (okuzan) [inline src/main/java/com/example/store/model/Category.java]:
```suggestion
@SQLDelete(sql = "UPDATE categories SET is_deleted = TRUE WHERE id = ?")
@Where(clause = "is_deleted = FALSE")
```
true and false are sql keywords too

File: src/main/java/com/example/store/model/User.java [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 @Data
 @Entity
-@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@SQLDelete(sql = "UPDATE users SET is_deleted = true WHERE id=?")
```
Comment (okuzan) [inline src/main/java/com/example/store/model/User.java]:
```suggestion
@SQLDelete(sql = "UPDATE users SET is_deleted = TRUE WHERE id = ?")
```

File: src/main/java/com/example/store/controller/CategoryController.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category management", description = "Endpoints to managing categories")
+@RestController
+@RequestMapping(value = "/category")
```
Comment (okuzan) [inline src/main/java/com/example/store/controller/CategoryController.java]:
```suggestion
@RequestMapping(value = "/categories")
```

File: src/main/java/com/example/store/repository/book/BookRepository.java [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
+import java.util.List;
 import java.util.Optional;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
     Optional<Book> findBookByIsbn(String isbn);
+
+    List<Book> findAllByCategoriesId(Long id, Pageable pageable);
```
Comment (okuzan) [inline src/main/java/com/example/store/repository/book/BookRepository.java]:
```suggestion
    List<Book> findAllByCategoryId(Long id, Pageable pageable);
```

File: src/main/resources/db/changelog/changes/create-books_categories-table.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            referencedTableName: books
+            referencedColumnNames: id
+            constraintName: FK_books_categories_books
+
+        - addForeignKeyConstraint:
+            baseTableName: books_categories
+            baseColumnNames: category_id
+            referencedTableName: categories
+            referencedColumnNames: id
+            constraintName: FK_books_categories_categories
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/create-books_categories-table.yaml:32]:
```suggestion
            constraintName: FK_books_categories_categories
            
```

File: src/main/java/com/example/store/service/impl/CategoryServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    @Override
+    public void deleteById(Long id) {
+        categoryRepository.deleteById(id);
+    }
+
+    @Override
+    public CategoryDto update(Long id, CreateCategoryRequestDto requestDto) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find category by id: " + id)
+        );
```
Comment (okuzan) [inline src/main/java/com/example/store/service/impl/CategoryServiceImpl.java]: let's create
private Category getCategoryById(Long id) {
    return categoryRepository.findById(id)
            .orElseThrow(() -> new EntityNotFoundException("Can't find category by id: " + id));
}

File: src/main/java/com/example/store/service/impl/CategoryServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> getBooksByCategory(Long id, Pageable pageable) {
+        List<BookDtoWithoutCategoryIds> allByCategoriesId =
+                bookRepository.findAllByCategoriesId(id, pageable).stream()
+                .map(bookMapper::toBookDtoWithoutCategoryIds)
+                .toList();
+        if (allByCategoriesId.isEmpty()) {
+            throw new EntityNotFoundException("Can't find category in db");
```
Comment (okuzan) [inline src/main/java/com/example/store/service/impl/CategoryServiceImpl.java]: include id in the message

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]

**Pull request merged**